src/dynamics/prognostic_variables.jl

```diff
@@ -1,411 +1,407 @@
-    set_pressure!(progn, spectral(pressure, M.spectral_transform); lf)
-
-"""
-    set_pressure!(progn::PrognosticVariables{NF},
-                  pressure::AbstractGrid,
-                  lf::Integer=1) where NF
+# set vor_div <- grid 
+function set_vordiv!(vor::LowerTriangularArray, div::LowerTriangularArray, u::AbstractGridArray, v::AbstractGridArray, geometry::Geometry, S::Union{Nothing, SpectralTransform}=nothing; add::Bool, coslat_scaling_included::Bool=false)
+    
+    u_ = coslat_scaling_included ? u : RingGrids.scale_coslat⁻¹(u)
+    v_ = coslat_scaling_included ? v : RingGrids.scale_coslat⁻¹(v)
 
-Sets the prognostic variable with the surface pressure in grid space at leapfrog index `lf`.
-"""
-set_pressure!(progn::PrognosticVariables, pressure::AbstractGrid; lf::Integer=1) =
-    set_pressure!(progn, spectral(pressure, one_more_degree=true); lf)
+    u_spec = isnothing(S) ? transform(u_) : transform(u_, S)
+    v_spec = isnothing(S) ? transform(v_) : transform(v_, S)
 
-"""
-    set_pressure!(progn::PrognosticVariables{NF},
-                  pressure::AbstractMatrix,
-                  Grid::Type{<:AbstractGrid},
-                  lf::Integer=1) where NF
+    set_vordiv!(vor, div, u_spec, v_spec, geometry, S; add, coslat_scaling_included=true)
+end 
 
-Sets the prognostic variable with the surface pressure in grid space at leapfrog index `lf`.
-"""
-set_pressure!(progn::PrognosticVariables, pressure::AbstractMatrix; lf::Integer=1,
-    Grid::Type{<:AbstractGrid}=FullGaussianGrid) = set_pressure!(progn, spectral(pressure; Grid, one_more_degree=true); lf)
+# set vor_div <- LTA
+function set_vordiv!(vor::LowerTriangularArray, div::LowerTriangularArray, u::LowerTriangularArray, v::LowerTriangularArray, geometry::Geometry, S::Union{Nothing, SpectralTransform}=nothing; add::Bool, coslat_scaling_included::Bool=false) 
   
-"""
-    get_var(progn::PrognosticVariables, var_name::Symbol; lf::Integer=1)
-
-Returns the prognostic variable `var_name` at leapfrog index `lf` as a `Vector{LowerTriangularMatrices}`.
-"""
-function get_var(progn::PrognosticVariables, var_name::Symbol; lf::Integer=1)
-    @assert has(progn, var_name) "PrognosticVariables has no variable $var_name"
-    return [getfield(layer.timesteps[lf], var_name) for layer in progn.layers]
+    S = isnothing(S) ? SpectralTransform(geometry.spectral_grid) : S
+     
+    u_ = coslat_scaling_included ? u : transform(RingGrids.scale_coslat⁻¹(transform(u, S)), S)
+    v_ = coslat_scaling_included ? v : transform(RingGrids.scale_coslat⁻¹(transform(u, S)), S)
+
+    if size(vor) != size(u_) != size(v_)
+        u_new = zero(vor)
+        copyto!(u_new, u_) 
+
+        v_new = zero(vor)
+        copyto!(v_new, v_)
+
+        curl!(vor, u_new, v_new, S; add)
+        divergence!(div, u_new, v_new, S; add)
+    else 
+        curl!(vor, u_, v_, S; add)
+        divergence!(div, u_, v_, S; add)
```

Comment:
there's a division by the radius missing here. As discussed somewhere in the docs (sufficiently I hope) all spectral gradient operators omit the radius scaling as it's not needed in the dynamical core (hence we use vor, div being scaled by the radius). I've therefore added to a bunch of `curl` and `divergence` functions a `radius` keyword (default 1) that one needs to pass on. The `SpectralTransform` doesn't know about the radius (because it's invariant to the radius)

Reply:
All spectral gradients now can take an optional `radius` keyword argument. Otherwise `radius=1` is assumed.